### PR TITLE
You don't need that abstraction

### PR DESCRIPTION
**Pull request by mauroservienti**
## PoA

- [x] Synopsis
- [x] Tags
- [x] Header image https://unsplash.com/photos/2dkgXTfPfTg
- [x] Header image credits
- [x] Review
- [x] Rename file to match publish date
- [ ] Buffer content

### CODE REVIEW

File: _posts/you-dont-need-that-abstraction.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+---
+layout: post
+header_image: /img/posts/you-dont-need-that-abstraction/header.jpg
+title: "You don't need that abstraction"
+author: Mauro Servienti
+synopsis: "Abstractions, abstractions everywhere! We're obsessed with clean design and architecture. Do we need that, or the introduced cognitive load comes with more issues than benefits?"
```
Comment (kbaley) [inline _posts/you-dont-need-that-abstraction.md:6]:
```suggestion
synopsis: "Abstractions, abstractions everywhere! We're obsessed with clean design and architecture. Do we need that, or does the introduced cognitive load come with more issues than benefits?"
```

File: _posts/you-dont-need-that-abstraction.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+---
+
+One of the beauties of service-oriented architecture (SOA) is simplicity.
+
+## WAT? Aren't distributed systems more complex by definition?
+
+Well, yes. But also no. In my experience, overall, a distributed system comes with slight complexity differences compared to a monolithic architecture. It's where the complexity is that is important and dramatically changes the day-to-day experience.
+
+With a monolithic architecture, the entire system is at our disposal. Adding a reference to a project in the solution is trivial, and our preferred IDE IntelliSense will immediately show new types and new members. 
+That is like honey for bees: it's hard to resist temptation. It's effortless to steer away from a well-designed monolithic architecture.
```
Comment (kbaley) [inline _posts/you-dont-need-that-abstraction.md:19]:
```suggestion
With a monolithic architecture, the entire system is at our disposal. Adding a reference to a project in the solution is trivial, and our preferred IDE IntelliSense will immediately show new types and new members. That is like honey for bees: it's hard to resist temptation. It's effortless to steer away from a well-designed monolithic architecture.
```

File: _posts/you-dont-need-that-abstraction.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+Also, autonomous components drive more adherence to the single responsibility principle, and [NServiceBus message handlers](https://docs.particular.net/nservicebus/handlers/) are an excellent example. It means that the component can do one thing at a time. There is no simple workaround. It's much easier and riskier to add a method to a class in a monolithic architecture rather than adding additional behaviors to an autonomous component in a distributed system. There is so much more friction that we don't even try.
+
+Finally, code simplicity comes with less cognitive load.
+
+However, distributed systems come with their own set of complexity. Monitoring and debugging are representative samples of things that are harder than in monolithic architecture.
```
Comment (kbaley) [inline _posts/you-dont-need-that-abstraction.md:49]: This is a bit abrupt. You spend the bulk of the post debunking the myth that SOA is complicated, then admit that certain aspects of it are hard without commenting on it further. Is it so hard that it shifts the equation significantly? Are you going to expand on it in a future post?

Comment (mauroservienti) [reply _posts/you-dont-need-that-abstraction.md]: Good point, reworded.

**Pull request merged**